.github/workflows/verify-app-build.yml

```diff
@@ -0,0 +1,255 @@
+name: Build App
+
+#######
+# Verify building an app for a target platform, format, and framework on a particular OS.
+# If a platform and format are not specified, then an app is built for each platform and format supported on the OS.
+#######
+
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        description: 'Python version to use; defaults to latest Python release.'
+        default: '3.X'
+        type: string
+      python-source:
+        description: 'Where to install python from; i.e. github or winstore.'
+        default: 'github'
+        type: string
+      runner-os:
+        description: 'The OS to use to build the App; must be a fully qualified GitHub runner OS, e.g. ubuntu-latest.'
+        required: true
+        type: string
+      framework:
+        description: 'Framework to use to build the App, e.g. toga.'
+        required: true
+        type: string
+      repository:
+        description: 'GitHub repository to checkout; defaults to repo running this workflow.'
+        default: ${{ github.repository }}
+        type: string
+      target-platform:
+        description: 'The target platform for the app, i.e. linux, macos, or windows. Leave blank all supported platforms.'
+        default: ''
+        type: string
+      target-format:
+        description: 'The target format for the app, e.g. appimage, xcode, app, etc. Leave blank all supported formats.'
+        default: ''
+        type: string
```

Comment:
To clarify my understanding: there wouldn't be a way to specify "macOS app and Windows VisualStudio", because "target-platform" would be "macos, windows" and target-format would be "app, VisualStudio", which would then catch "Windows app" - correct? I don't think this is a dealbreaker - in practice, we're either doing everything, or doing one specific platform - but it's worth being explicit. The only real alternative I can think of would to be make target-platform a single string, rather than a list; (so - target-platform, but target-format*s*), and then do an exact match rather than a contains when checking target-platform.

Reply:
Currently, this script does not support multiple values for `target-platform` or `target-format` (or any of the other inputs for that matter).

This script is a combination of the use-cases from Briefcase and the template repos. Briefcase builds as many Apps as are possible for a framework and OS combo. The template repos only build a specific platform/format at any one time.

So, to satisfy these two use-cases, you can specify a specific target platform/format and build that while also allowing a simpler matrix based only on OS and framework while building all platform/format combos.

So, at the end of the day, this workflow is designed to to be used in a matrix that minimally defines an OS and a framework.

Reply:
Got it - I misread/misunderstood how the `fromJSON` matching was working.